Jellyfin.Server.Implementations/JellyfinDbProvider.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using Jellyfin.Data.DbContexts;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Jellyfin.Server.Implementations
+{
+    /// <summary>
+    /// Acts as an intermediate layer between the application's service provider and
+    /// the rest of the application to provide access to a connection pool to the
+    /// underlying database.
```

Comment:
Fair enough. When I originally documented this the underlying `DbContext` service provider was using `AddDbContextPool`. This works by reusing instances within DI scopes, so I'm not sure if it is even possible to reimplement given that the same endpoint can be called from different threads simultaneously. This is definitely more accurate now

Reply:
Ah okay, definitely makes more sense in the context of `AddDbContextPool`.

I _think_ what happen to happen to get scoped requests working correctly is that over time is we need to switch over services to register them as 'scoped' instead of 'singleton'. That way a separate new instance of each service will be constructed for each request and can safely have `DbContext` injected directly via the constructor. A project for another day haha

Reply:
From my understanding, scoping means to define separate sections of code where instances are reused within those sections. What you seem to be describing sounds more like a transient service to me, but I may be mistaken.

Reply:
I may also be mistaken but here is my understanding:

'Scoped' means that a new instance is created once per HTTP request. Generally the pattern is to register the DbContext and your controllers/services as scoped. This way, each request has it's own set of controllers/services and they all share the same `DbContext` _within that request scope_. So if you have two requests to the same controller in parallel, then two different instances of the controller and DbContext are created. This way you can share resources for the request, but also have thread safety, since each request runs on a single thread